setup.py

```diff
@@ -1,5 +1,32 @@
+import sys
 import setuptools
 
+
+# Taken from Numba
+def _guard_python_version(max_python):
```

Comment:
@henryiii in retrospect while thinking on https://github.com/iris-hep/qastle/pull/72#discussion_r1529710773, this might not be worth adding as `qastle` is pure-Python and so while Python that is newer than `max_python` would fail to run in 

```
python -m build .
```

once the wheel is built (on say an older Python) there's nothing keeping that pure-Python wheel from getting installed on a Python that is `max_python` or newer, unlike `numba` where it has wheels with compiled extensions. So I guess `_guard_python_version` can just be removed.

Reply:
Yes, that's true - I'd leave it off. I'd either let whatever fails fail (you'll quickly get a report I'd expect letting you know what needs to be fixed!), or add a check and error in `__init__.py`.

Reply:
(also, checking on pre-release versions is a good idea, beat your users to the punch. ;) )

Reply:
For future reference for myself, c.f. https://packaging.pypa.io/en/latest/version.html